oak_channel/src/frame.rs

```diff
@@ -116,8 +119,8 @@ impl Framed {
             let body_length: usize = length
                 .checked_sub(BODY_OFFSET)
                 .expect("body length underflow");
-            let mut body: Vec<u8> = vec![0; body_length];
-            self.inner.read(&mut body)?;
+            let body = allocator(body_length);
```

Comment:
FWIW this seems to me the canonical use case for https://docs.rs/bytes/latest/bytes/ . Have you considered that? You could allocate a single `BytesMut::with_capacity` and pass that around, then directly call `put` from here.

Reply:
I poked around that crate a bit. On paper, yes, it should be a good fit, but honestly I felt it isn't an improvement over just using the slices -- so I'd like to submit this as-is, as it's still an improvement.

Reply:
Personally I find the back and forth with allocator function not very intuitive, I had to re-read a couple of times to figure out what it was trying to allocate and where. I think Bytes is the way to go, anyways please go ahead with this and I'll create another issue to track that.

Reply:
I think you could also just use a vec with a large enough capacity, and pass that around as a mut ref and then use extend_from_slice. Or use the helper trait from https://docs.rs/bytes/latest/bytes/trait.BufMut.html (without having to use BytesMut -- it should work on a regular Vec too).